lib/SGN/Controller/AJAX/Trial.pm

```diff
@@ -1247,115 +1269,137 @@ sub upload_multiple_trial_designs_file_POST : Args(0) {
+    for my $trial (values%$parsed_data) {
+        if (exists $trial->{'breeding_program'}) {
+            $breeding_program_name = $trial->{'breeding_program'};
         }
+    }
+    unless ($breeding_program_name) {
+        $c->stash->{rest} = {errors => "Breeding program not found in the uploaded file."};
+        return;
+    }
+    print STDERR "Breeding program name: $breeding_program_name\n";
 
-        # save entry numbers, if provided
-        if ( $entry_numbers && scalar(keys %$entry_numbers) > 0 && $current_save->{'trial_id'} ) {
-            my %entry_numbers_prop;
-            my @stock_names = keys %$entry_numbers;
-
-            # Convert stock names from parsed trial template to stock ids for data storage
-            my $stocks = $chado_schema->resultset('Stock::Stock')->search({ uniquename=>{-in=>\@stock_names} });
-            while (my $s = $stocks->next()) {
-                $entry_numbers_prop{$s->stock_id} = $entry_numbers->{$s->uniquename};
-            }
-
-            # Lookup synonyms of accession names
-            my $synonym_cvterm_id = SGN::Model::Cvterm->get_cvterm_row($chado_schema, 'stock_synonym', 'stock_property')->cvterm_id();
-            my $acc_synonym_rs = $chado_schema->resultset("Stock::Stock")->search({
-                'me.is_obsolete' => { '!=' => 't' },
-                'stockprops.value' => { -in => \@stock_names},
-                'stockprops.type_id' => $synonym_cvterm_id
-            },{join => 'stockprops', '+select'=>['stockprops.value'], '+as'=>['synonym']});
-            while (my $r=$acc_synonym_rs->next) {
-                if ( exists($entry_numbers->{$r->get_column('synonym')}) ) {
-                    $entry_numbers_prop{$r->stock_id} = $entry_numbers->{$r->get_column('synonym')};
-                }
-            }
-
-            # store entry numbers
-            my $trial = CXGN::Trial->new({ bcs_schema => $chado_schema, trial_id => $current_save->{'trial_id'} });
-            $trial->set_entry_numbers(\%entry_numbers_prop);
-        }
-      }
+    # my $projects = CXGN::BreedersToolbox::Projects->new({ schema => $chado_schema });
+    # my $breeding_program = $projects->get_breeding_program_by_name($breeding_program_name);
 
-    };
+    my %all_designs = %{$parsed_data};
+    my %save;
+    $save{'errors'} = [];
 
-    try {
-        $chado_schema->txn_do($coderef);
-    } catch {
-        print STDERR "Transaction Error: $_\n";
-        push @{$save{'errors'}}, $_;
-    };
+    # print STDERR "breeding_program_name:  $breeding_program_name \n";
+    # print STDERR "infile:  $infile \n";
+
+    my $async_upload = CXGN::Tools::Run->new();
+    $async_upload->run_async("perl $basepath/bin/upload_multiple_trial_design.pl -H $dbhost -D $dbname -P $dbpass -w $basepath -U $dbuser -b $breeding_program_name -i $infile -un $username -e $email_address -eo $email_option_enabled -r $temp_file_nd_experiment_id");
+
+    # my $coderef = sub {
```

Comment:
where was this code moved? if it shouldn't be here then I'd remove it instead of commenting out

Reply:
@nmenda, thanks for the remainder. Sorry, it was even mentioned in previous meetings, and missed it. Now it is fixed.